src/modules/mim/firewall.json

```diff
@@ -1,41 +1,187 @@
+              },
+              {
+                "name": "ruleType",
+                "schema": {
+                  "type": "enum",
+                  "valueSchema": "integer",
+                  "enumValues": [
+                    {
+                      "name": "inbound",
+                      "enumValue": 0
+                    },
+                    {
+                      "name": "outbound",
+                      "enumValue": 1
+                    },
+                    {
+                      "name": "forward",
+                      "enumValue": 2
+                    }
+                  ]
+                }
+              },
+              {
+                "name": "protocol",
+                "schema": {
+                  "type": "enum",
+                  "valueSchema": "integer",
+                  "enumValues": [
+                    {
+                      "name": "all",
+                      "enumValue": 0
+                    },
+                    {
+                      "name": "tcp",
+                      "enumValue": 1
+                    },
+                    {
+                      "name": "udp",
+                      "enumValue": 2
+                    },
+                    {
+                      "name": "icmp",
+                      "enumValue": 3
+                    }
+                  ]
+                }
+              },
+              {
+                "name": "source",
+                "schema": {
+                  "type": "object",
+                  "fields": [
+                    {
+                      "name": "address",
+                      "schema": "string"
+                    },
+                    {
+                      "name": "port",
+                      "schema": "integer"
+                    },
+                    {
+                      "name": "interface",
+                      "schema": "string"
+                    }
+                  ]
+                }
+              },
+              {
+                "name": "destination",
+                "schema": {
+                  "type": "object",
+                  "fields": [
+                    {
+                      "name": "address",
+                      "schema": "string"
+                    },
+                    {
+                      "name": "port",
+                      "schema": "integer"
+                    },
+                    {
+                      "name": "interface",
+                      "schema": "string"
+                    }
+                  ]
+                }
+              },
+              {
+                "name": "state",
+                "schema": {
+                  "type": "array",
+                  "elementSchema": {
+                    "type": "enum",
+                    "valueSchema": "string"
+                  }
+                }
+              }
+            ]
+          }
+        },
+        {
+          "name": "deleteRule",
```

Comment:
>deleteRule [](http://example.com/codeflow?start=19&length=10)

Rather than a separate object, have an action Add/Remove on the desiredFirewallRule